cai_causal_graph/graph_components.py

```diff
@@ -23,6 +24,65 @@
 from cai_causal_graph.type_definitions import EDGE_T, NodeLike, NodeVariableType
 
 
+def get_variable_name_and_lag(node_name: NodeLike) -> Tuple[str, int]:
+    """
+    Extract the variable name from a node name.
+
+    Example:
+        'X lag(n=2)' -> 'X' if lagged in the past,
+        'X future(n=2)' -> 'X' if lagged in the future.
+    """
+    # get the string name if the node is a Node object
+    if isinstance(node_name, Node):
+        node_name = node_name.identifier
+    assert isinstance(node_name, str), f'Invalid node name: {node_name}.'
+    match = re.match(r'(\w+) lag\(n=(\d+)\)', node_name)
+
+    match = re.match(r'(\w+)(?: lag\(n=(\d+)\))?(?: future\(n=(\d+)\))?', node_name)
+
+    if match:
+        variable_name = match.group(1)
+        past_lag = match.group(2)
+        future_lag = match.group(3)
+
+        if past_lag:
+            return variable_name, -int(past_lag)
```

Comment:
Just double checking the sign as this seems like it will be -2 for lag=2 but in `get_name_from_lag` we show negative for future lag and positive for past lag.

Reply:
the docstring was wrong in `get_name_from_lag`